tools/extra/mmlink/remote_dbg/remote_dbg.h

```diff
@@ -0,0 +1,42 @@
+// Copyright(c) 2020, Intel Corporation
+//
+// Redistribution  and  use  in source  and  binary  forms,  with  or  without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of  source code  must retain the  above copyright notice,
+//   this list of conditions and the following disclaimer.
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+// * Neither the name  of Intel Corporation  nor the names of its contributors
+//   may be used to  endorse or promote  products derived  from this  software
+//   without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+// IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+// LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+// CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+// SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+#include <cstdint>
+#include <cstdio>
+
+#define PRINT_ERR(format, ...) \
+	printf("%s:%u:%s() : " format "\n", __FILE__, __LINE__, __func__,\
+               ## __VA_ARGS__)
+
+
+class remote_dbg
+{
+public:
+  remote_dbg(){}
```

Comment:
The compiler can handle c++11 syntax now.  I recommend using the more expressive syntax and complete the full interface to prevent incorrect usage.  For example,

class remote_dbg
{
public:
  remote_dbg() = default;
  remote_dbg(const remote_dbg&) = delete;
  remote_dbg& operator=(const remote_dbg&) = delete;
  virtual ~remote_dbg() = default;
  
  virtual int run(volatile uint64_t *mmio, const char *address, int port) = 0;

};

Reply:
good call, will update.